compiler/noirc_frontend/src/parser/parser/traits.rs

```diff
@@ -593,4 +591,11 @@ mod tests {
         assert_eq!(noir_trait.items.is_empty(), noir_trait_alias.items.is_empty());
         assert!(!noir_trait.is_alias);
     }
+
+    #[test]
+    fn parse_trait_with_constant_missing_semicolon() {
+        let src = "trait Foo { let x: Field = 1 }";
+        let (_, errors) = parse_program_with_dummy_file(src);
+        assert!(!errors.is_empty());
+    }
```

Comment:
Do we have a similar test for associated types?

Reply:
Good point, no. I just added some.